From cornelia.huck@de.ibm.com Tue Apr 10 05:33:12 2007
From: Cornelia Huck <cornelia.huck@de.ibm.com>
Date: Tue, 10 Apr 2007 14:35:27 +0200
Subject: kobject: kobject_add() reference leak
To: Greg K-H <greg@kroah.com>
Cc: WANG Cong <xiyou.wangcong@gmail.com>, linux-kernel@vger.kernel.org, Andrew Morton <akpm@linux-foundation.org>, viro@zeniv.linux.org.uk
Message-ID: <20070410143527.2b7718b8@gondolin.boeblingen.de.ibm.com>


We leak a reference if we attempt to add a kobject with no name.

Signed-off-by: Cornelia Huck <cornelia.huck@de.ibm.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 lib/kobject.c |    1 +
 1 file changed, 1 insertion(+)

--- a/lib/kobject.c
+++ b/lib/kobject.c
@@ -174,6 +174,7 @@ int kobject_shadow_add(struct kobject * 
 	if (!*kobj->k_name) {
 		pr_debug("kobject attempted to be registered with no name!\n");
 		WARN_ON(1);
+		kobject_put(kobj);
 		return -EINVAL;
 	}
 	parent = kobject_get(kobj->parent);
